WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -543,42 +543,34 @@ class MediaCoordinator: NSObject {
     }
 
     /// Encapsulates an observer block and an optional observed media item or post.
-    struct MediaObserver {
-        let media: Media?
-        let post: AbstractPost?
+    private struct MediaObserver {
+        let mediaObjectID: NSManagedObjectID?
+        let postObjectID: NSManagedObjectID?
         let onUpdate: ObserverBlock
 
-        init(onUpdate: @escaping ObserverBlock) {
-            self.media = nil
-            self.post = nil
-            self.onUpdate = onUpdate
-        }
-
         init(media: Media?, onUpdate: @escaping ObserverBlock) {
-            self.media = media
-            self.post = nil
+            self.mediaObjectID = media?.objectID
+            self.postObjectID = nil
             self.onUpdate = onUpdate
         }
 
         init(post: AbstractPost, onUpdate: @escaping ObserverBlock) {
-            self.media = nil
-            self.post = post
+            self.mediaObjectID = nil
+            self.postObjectID = post.objectID
             self.onUpdate = onUpdate
         }
     }
 
     /// Utility method to return all observers for a specific media item,
     /// including any 'wildcard' observers that are observing _all_ media items.
     ///
-    private func observersForMedia(_ media: Media) -> [MediaObserver] {
-        let mediaObservers = self.mediaObservers.values.filter({ $0.media?.uploadID == media.uploadID })
+    private func observersForMedia(withObjectID mediaObjectID: NSManagedObjectID, originalPostIDs: [NSManagedObjectID]) -> [MediaObserver] {
+        let mediaObservers = self.mediaObservers.values.filter({ $0.mediaObjectID == mediaObjectID })
```

Comment:
The [`uploadID`](https://github.com/wordpress-mobile/WordPress-iOS/blob/8d84b2468e991f037f544086267ce8269930ef06/WordPress/Classes/Services/MediaCoordinator.swift#L740-L742) is `objectID` as a string, so comparing `NSManagedObjectID` should be the same as comparing `uploadID`.